module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1835,6 +1835,12 @@ zfs_fhtovp(vfs_t *vfsp, fid_t *fidp, int flags, vnode_t **vpp)
 		return (SET_ERROR(EINVAL));
 	}
 
+	if (fid_gen > 1 || setgen != 0) {
```

Comment:
This predictably failed to build since setgen is declared inside the if () scope above.  It is indeed not used now and should be always zero, but this check is broken.
Check for fid_gen > 1 looks just wrong to me. I am not sure why it was added here, since it can be 0 for snapshots and must be non-zero for normal dataset files.

Reply:
It was added on the Linux side as an additional sanity check as part of 9b77d1c9585572b7ce3af204d40278752d5f5842, which was for a Linux specific issue.  I see `setgen` is similarly unused in the Illumos code.  So the question is, is there value in a check along these lines for FreeBSD or would you just rather entirely remove the unused code?

Reply:
I think checking setgen would be good.  It sound potentially dangerous if the same file can be addressed via different file handles.